server/aesdsocket.c

```diff
@@ -0,0 +1,154 @@
+
+
+// Used http://www.cs.cmu.edu/~prs/15-441-F10/lectures/r01-sockets.pdf as reference
+// for setting up sockets.
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <signal.h>
+#include <syslog.h>
+
+#define SERVER_PORT 9000
+#define BUFFER_SIZE 1024
+
+int sockfd, connfd;
+
+void signal_handler(int signal) {
+    if (signal == SIGINT || signal == SIGTERM) {
+        // Close the socket
+        close(sockfd);
+        // Delete the file
+        remove("/var/tmp/aesdsocketdata");
+        // Log the message
+        syslog(LOG_INFO, "Caught signal, exiting");
+        // Close syslog
+        closelog();
+        exit(0);
```

Comment:
@tiba6275 are all of these actions safe to do from signal handler?  Consider a design where you can handle cleanup in the main loop instead.